devices/Mcp7940xx/Alarm.cs

```diff
@@ -0,0 +1,286 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.I2c;
+using System.Globalization;
+using Iot.Device.Common;
+
+namespace Iot.Device.Mcp7940xx
+{
+    /// <summary>
+    /// I2C Real-Time Clock/Calendar with SRAM.
+    /// </summary>
+    public partial class Mcp7940m
+    {
+        /// <summary>
+        /// Represents an alarm  on a Mcp7940xx series device.
+        /// </summary>
+        public class Alarm
+        {
+            private byte _second;
+            private byte _minute;
+            private byte _hour;
+            private byte _day;
+            private byte _month;
+
+            private byte Clamp(byte value, byte low, byte high)
```

Comment:
Since we are embedded and memory is a premium, I wasn't sure if it was worth it to reference another assembly the user might not be making use of for just one function. I can switch it over easily though, let me know.